code/script.js

```diff
@@ -1,47 +1,228 @@
 // All the DOM selectors stored as short variables
-const chat = document.getElementById('chat')
+const chat = document.getElementById("chat");
+const nameInput = document.getElementById("name-input");
+const submitBtn = document.getElementById("btn");
+const inputWrapper = document.getElementById("input-wrapper");
 
+//[updated] 'name' is reserved as global variable? instead 'name' -> 'username'
+let username;
 // Global variables, if you need any, declared here
 
 // Functions declared here
 
 // This function will add a chat bubble in the correct place based on who the sender is
 const showMessage = (message, sender) => {
-  if (sender === 'user') {
+  if (sender === "user") {
     chat.innerHTML += `
       <section class="user-msg">
         <div class="bubble user-bubble">
           <p>${message}</p>
         </div>
         <img src="assets/user.png" alt="User" />  
       </section>
-    `
-  } else if (sender === 'bot') {
+    `;
+  } else if (sender === "bot") {
     chat.innerHTML += `
       <section class="bot-msg">
         <img src="assets/bot.png" alt="Bot" />
         <div class="bubble bot-bubble">
           <p>${message}</p>
         </div>
       </section>
-    `
+    `;
   }
   // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
-  chat.scrollTop = chat.scrollHeight
-}
+  chat.scrollTop = chat.scrollHeight;
+};
+
+// Restart function
+const restartPage = (botMessage, buttonMessage) => {
+  inputWrapper.innerHTML = `
+  <button id="restart">${buttonMessage}</button>
+   `;
+  showMessage(botMessage, "bot");
+
+  document.getElementById("restart").addEventListener("click", () => {
+    window.location.reload(); //great!!
+  });
+};
 
 // Starts here
 const greeting = () => {
-  showMessage(`Hello there, What's your name?`, 'bot')
+  showMessage(`Hello there, What a lovely day🍎 What is your name?`, "bot");
   // Just to check it out, change 'bot' to 'user' here 👆
-}
+};
 
-// Set up your eventlisteners here
+//[updated] function name updated from handleNameInput to getUsername
+const getUsername = (event) => {
+  event.preventDefault();
+  username = nameInput.value; //[updated] assigned username global variable here
+  showMessage(username, "user");
+
+  //[updated] if username is empty (null), ask name again
+  if (!username) {
+    setTimeout(greeting, 1000);
+  } else {
+    setTimeout(showFoodOptions, 1000);
+  }
+  nameInput.value = "";
+};
```

Comment:
Nice thing to not forget about this and what will happen if the user don't put in a name. I tried it and it works well, the user get asked the initial question again. Really nice (we totally forgot this for our chat bot so if you don't mind I'm saving this little snippet of code for the future, might come in handy ;))

Reply:
So true! Agree with Camilla here. You could add this logic to the rest of the questions but that's totally optional of course. :) 👍